internal/backup/configs/tarball/backup.go

```diff
@@ -0,0 +1,71 @@
+package tarball
+
+import (
+	"archive/tar"
+	"fmt"
+	"io"
+	"path"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/yaml"
+)
+
+type Backup struct {
+	mu     sync.Mutex
+	writer *tar.Writer
+}
+
+type BackupResourcesFilter interface {
+	Matches(object runtime.Object) bool
+}
+
+func NewBackup(sink io.Writer) *Backup {
+	return &Backup{
+		writer: tar.NewWriter(sink),
+	}
+}
+
+func (b *Backup) PutObject(object runtime.Object) error {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	metadataAccessor, err := meta.Accessor(object)
+	if err != nil {
+		return fmt.Errorf("%w: %s does not contain metadata to filter with", err, object.GetObjectKind().GroupVersionKind().String())
+	}
+
```

Comment:
I think we can strip  managedFields, as they are not useful:
```suggestion
	}
	metadataAccessor.SetManagedFields(nil)

```